src/khoj/database/adapters/__init__.py

```diff
@@ -416,6 +416,11 @@ def get_all_users() -> BaseManager[KhojUser]:
     return KhojUser.objects.all()
 
 
+def check_valid_user(user: KhojUser | None):
```

Comment:
Just fruit for thought, it might be worth making this a decorator? Seems like a neater way to handle top-level function validation that we're standardizing across the code. Non-blocking, up to you.

Reply:
Yes